src/librustc/session/config.rs

```diff
@@ -1852,8 +1852,11 @@ pub fn rustc_optgroups() -> Vec<RustcOptGroup> {
 }
 
 // Convert strings provided as --cfg [cfgspec] into a crate_cfg
-pub fn parse_cfgspecs(cfgspecs: Vec<String>) -> FxHashSet<(String, Option<String>)> {
-    syntax::with_globals(move || {
+pub fn parse_cfgspecs(
+    driver_symbols: &[&str],
+    cfgspecs: Vec<String>,
+) -> FxHashSet<(String, Option<String>)> {
+    syntax::with_globals(driver_symbols, move || {
```

Comment:
I'm not sure this one needs the extra symbols.
It's a "leaf" function that shouldn't delegate to any tool-specific functionality.

Reply:
It should not need any symbols at all I believe. Would you mind if I did that change in a separate PR?

Reply:
It needs symbols because it parses paths and therefore needs to use at least `self`/`super`/etc.
I guess it's ok to leave this as is because I'm not *entirely* sure that parsing cannot be intercepted by a tool now or in the future.